### PR TITLE
Clarify documentation related to recorders and recording backends 

### LINKED ISSUE #3070: Documentation for recording spike times is obtuse
**Describe the flaw**
The main `spike_recorder` [documentation page](https://nest-simulator.readthedocs.io/en/latest/models/spike_recorder.html) doesn't mention that there is an 'events' attribute containing the spike times. I had to click through two times to backends to [find it here](https://nest-simulator.readthedocs.io/en/latest/models/recording_backend_memory.html). 

**Suggest an improvement**
Describe the attribute on the main docs page

**Supporting materials**
[https://nest-simulator.readthedocs.io/en/latest/models/spike_recorder.html](https://nest-simulator.readthedocs.io/en/latest/models/spike_recorder.html)

### PR DESCRIPTION
**Pull request by jessica-mitchell**
Fixes #3070

This PR tries to clarify how to get data out from recording backends, and generally make the documentation more discoverable  between recorders and recording backends.

Explicit example of getting "events" out of the memory backend is added to each recorder device, so users don't need to click through several pages to find this information. The backend documentation pages are added directly to the guide for recording from simulations, along with more explicit examples, links to some examples are added as well.

See the output on Read the Docs: https://nest-simulator--3159.org.readthedocs.build/en/3159/models/spike_recorder.html
https://nest-simulator--3159.org.readthedocs.build/en/3159/devices/record_from_simulations.html#recording-backends

### CODE REVIEW

**Review by WillemWybo — APPROVED** [review]
Very clear now! I approve :-)

File: nestkernel/recording_device.h [hunk 1]
```diff
@@ -59,9 +59,22 @@ Data handling
 +++++++++++++
 
 All recorded data is handed over to the recording backend, selected
-via the ``record_to`` property. More details on available backends and
-their properties can be found in the :ref:`guide to recording from
-simulations <recording_backends>`.
+via the ``record_to`` property::
+
+   >>> sr  = nest.Create("spike_recorder", params={"record_to":"ascii", "time_in_steps": False})
+   >>> mm = nest.Create("multimeter", 1, {"record_from": ["V_m", "g_ex"], "record_to": memory})
```
Comment (jessica-mitchell) [inline nestkernel/recording_device.h:65]:
```suggestion
   >>> mm = nest.Create("multimeter", 1, {"record_from": ["V_m", "g_ex"], "record_to": "memory"})
```

File: nestkernel/recording_device.h [hunk 2]
```diff
@@ -59,9 +59,27 @@ Data handling
...
+
+::
+
+   mm_events = mm.get("events")
+
+.. note::
+
+   The type of recording backend you choose may affect the efficiency of your simulation.
+   Increasingly larger simulations may benefit from the `ascii` or `sionlib` backend, as the `memory`
+   backend can become slow at larger scales.
```
Comment (terhorstd) [inline nestkernel/recording_device.h:79]:
```suggestion
   The  `memory` backend is ideal for interactive work, but can only be used for limited amount of data. Transferring data to disk later on may be slower than directly writing from the NEST kernel via `ascii` or `sionlib` backends.

   Large simulations with many threads may benefit from the `sionlib` backend, as the `ascii` backend opens many files which can be very time consuming on parallel file systems.
```

**Comment by jessica-mitchell** [comment]
After in-person discussions with NEST team, I merge this PR

**Pull request merged**